boot/nxboot/loader/boot.c

```diff
@@ -403,6 +440,42 @@ static int perform_update(struct nxboot_state *state, bool check_only)
   return OK;
 }
 
+#ifdef CONFIG_NXBOOT_COPY_TO_RAM
+int nxboot_ramcopy(void)
+{
+  int primary;
+  struct nxboot_img_header header;
+  ssize_t bytes;
+  static uint8_t *buf;
+  size_t size;
+
+  primary = flash_partition_open(CONFIG_NXBOOT_PRIMARY_SLOT_PATH);
+  if (primary < 0)
+    {
+      return ERROR;
+    }
+
+  get_image_header(primary, &header);
+  buf = malloc(header.size);
+  if (!buf)
+    {
+      return ERROR;
+    }
+
+  size = header.size - CONFIG_NXBOOT_HEADER_SIZE;
```

Comment:
`header.size` is already the size of the image excluding the header, therefore subtracting `CONFIG_NXBOOT_HEADER_SIZE` shouldn't be needed.

Reply:
@michallenc - good catch, thanks. I will have to see why my loaded app ran OK missing its last 512 bytes!

Reply:
> I will have to see why my loaded app ran OK missing its last 512 bytes!
It's just debug data.

Reply:
done